### PR TITLE
fix(eth): Correctly set tx and receipt hashes on Eth headers

### PR DESCRIPTION
**Pull request by calbera**
We were not setting the `TxHash` and `ReceiptsHash` field on the header, which caused inconsistent hashes of the block header.

### CODE REVIEW

File: eth/core/processor.go [hunk 1]
```diff
@@ -215,8 +216,20 @@ func (sp *StateProcessor) Finalize(
 	// We unlock the state processor to ensure that the state is consistent.
 	defer sp.mtx.Unlock()
 
-	// Now that we are done processing the block, we update the header with the consumed gas.
+	// Now that we are done processing the block, we update the header with the consumed gas,
+	// txHash, and receiptsHash.
 	sp.header.GasUsed = sp.gp.BlockGasConsumed()
+	hasher := trie.NewStackTrie(nil)
```
Comment (itsdevbear) [inline eth/core/processor.go:222]: this is for sure wrong.

**Review by itsdevbear — CHANGES_REQUESTED** [review]
.

**Review by transmissions12 — APPROVED** [review]
looks good

**Comment by itsdevbear** [comment]
Is this confirmed to be correct when compared to geth?

**Comment by calbera** [comment]
Yes, all the fields are set on the header before hash is taken:

```
header &{0xbd3d11974290fac97d251ba559d14197d39eb9023f7f0474c0c6db2dce969ffc 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347 0x0000000000000000000000000000000000000000 0x339372a43d78370e10bccc934ce315d2233d789aff94b0c17a6ed8bcd1676664 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 2 15503 4712388 0 1492242989 [215 131 1 6 0 132 103 101 116 104 135 103 111 49 46 55 46 51 133 108 105 110 117 120 0 0 0 0 0 0 0 0 59 103 169 80 55 233 60 196 103 166 167 237 202 184 75 217 57 174 98 206 160 232 145 184 237 6 181 26 109 2 41 217 100 157 34 227 232 238 92 155 7 93 78 116 91 115 190 239 229 109 52 232 124 94 79 130 61 21 218 162 57 78 69 21 0] 0x0000000000000000000000000000000000000000000000000000000000000000 [0 0 0 0 0 0 0 0] <nil> <nil>}
^Cheader &{0x3d25a97133f3d08006ce157eb88316ac5fb5eba05ce4d121179731ae77867a1c 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347 0x0000000000000000000000000000000000000000 0x339372a43d78370e10bccc934ce315d2233d789aff94b0c17a6ed8bcd1676664 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 2 15504 4712388 0 1492243004 [215 131 1 6 0 132 103 101 116 104 135 103 111 49 46 55 46 51 133 108 105 110 117 120 0 0 0 0 0 0 0 0 67 74 82 252 62 73 143 226 186 154 49 203 56 124 202 144 102 149 21 4 249 119 53 56 241 17 15 234 122 25 35 4 91 244 249 25 251 19 162 249 248 31 86 226 21 124 39 188 153 95 36 52 31 40 130 33 163 98 101 10 104 29 149 116 0] 0x0000000000000000000000000000000000000000000000000000000000000000 [0 0 0 0 0 0 0 0] <nil> <nil>}
```

**Comment by calbera** [comment]
Also, we are overriding the txHash set by host chain here https://github.com/berachain/polaris/blob/ae6dc90afcd7cf2ac09b9d18c4af98229d6cc89e/cosmos/x/evm/plugins/block/plugin.go#L113

**Pull request closed without merging**